lib/features/base/mixin/handle_error_mixin.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/core/error/set_error.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:model/error_type_handler/set_method_error_handler_mixin.dart';
+
+mixin HandleSetErrorMixin {
+  void handleSetErrors({
+    SetMethodErrors? notDestroyedError,
+    SetMethodErrors? notUpdatedError,
+    SetMethodErrors? notCreatedError,
+    Set<SetMethodErrorHandler>? notDestroyedHandlers,
+    Set<SetMethodErrorHandler>? notUpdatedHandlers,
+    Set<SetMethodErrorHandler>? notCreatedHandlers,
+    SetMethodErrorHandler? unCatchErrorHandler
+  }) {
+    if (notDestroyedError != null && notDestroyedError.isNotEmpty) {
+      notDestroyedError.entries
+        .map((e) => _handleInChain(e, notDestroyedHandlers))
+        .map((optionError) => _handleRemainedError(unCatchErrorHandler, optionError));
+    }
+
+    if (notCreatedError != null && notCreatedError.isNotEmpty) {
+      notCreatedError.entries
+        .map((e) => _handleInChain(e, notCreatedHandlers))
+        .map((optionError) => _handleRemainedError(unCatchErrorHandler, optionError));
+    }
+
+    if (notUpdatedError != null && notUpdatedError.isNotEmpty) {
+      notUpdatedError.entries
+        .map((e) => _handleInChain(e, notUpdatedHandlers))
+        .map((optionError) => _handleRemainedError(unCatchErrorHandler, optionError));
+    }
+  }
+
+  Option<MapEntry<Id, SetError>> _handleInChain(MapEntry<Id, SetError> setError, Set<SetMethodErrorHandler>? handlers) {
+    try {
+      handlers!.firstWhere((handler) => handler.call(setError));
+      return const None<MapEntry<Id, SetError>>();
+    } catch (e) {
+      logError('HandleSetErrorMixin::chainHandle(): [Exception] $e');
+      return Some<MapEntry<Id, SetError>>(setError);
```

Comment:
Can you explain this function ? What is Some ?

Reply:
`Option`(Maybe) represent the absence of a value. Instead of returning one object when a function succeeds and null when it fails, your function should instead return an instance of an `Option`. 
```dart
class Some<A> extends Option<A> {
  final A _a;
  const Some(this._a);
  A get value => _a;
  @override B fold<B>(B ifNone(), B ifSome(A a)) => ifSome(_a);
  @override bool operator ==(other) => other is Some && other._a == _a;
  @override int get hashCode => _a.hashCode;
}

class None<A> extends Option<A> {
  const None();
  @override B fold<B>(B ifNone(), B ifSome(A a)) => ifNone();
  @override bool operator ==(other) => other is None;
  @override int get hashCode => 0;
}
```
`Some`: if value is available
`None`: if value not available.
For this case, `_handleInChain` will handle an error in chain, if error is handle successfully by one of the handler, no error will be remained -> return None. Otherwise, Some will be returned when no handler can handle the error. Using Option make us better to work in map. We don't need to care about it is null or not (avoid null-check boilerplate).